gravitee-am-test/specs/management/password-policy.jest.spec.ts

```diff
@@ -0,0 +1,170 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from 'cross-fetch';
+import * as faker from 'faker';
+import { afterAll, beforeAll, expect } from '@jest/globals';
+import { requestAdminAccessToken } from '@management-commands/token-management-commands';
+import { createDomain, deleteDomain, startDomain } from '@management-commands/domain-management-commands';
+import { createIdp } from '@management-commands/idp-management-commands';
+import {
+  createPasswordPolicy,
+  deletePasswordPolicy,
+  getAllPasswordPolicies,
+  getAllPasswordPoliciesRaw,
+  getPasswordPolicy,
+  getPasswordPolicyRaw,
+  setPasswordPolicyDefault, updatePasswordPolicy,
+} from '@management-commands/password-policy-management-commands';
+
+global.fetch = fetch;
+
+let accessToken;
+let domain;
+
+beforeAll(async () => {
+  const adminTokenResponse = await requestAdminAccessToken();
+  accessToken = adminTokenResponse.body.access_token;
+  expect(accessToken).toBeDefined();
+
+  const createdDomain = await createDomain(accessToken, 'domain-idp', faker.company.catchPhraseDescriptor());
+  expect(createdDomain).toBeDefined();
+  expect(createdDomain.id).toBeDefined();
+
+  const domainStarted = await startDomain(createdDomain.id, accessToken);
+  expect(domainStarted).toBeDefined();
+  expect(domainStarted.id).toEqual(createdDomain.id);
+
+  domain = domainStarted;
+});
+
+describe('password policy management', () => {
+  it('must return no password policy', async () => {
+    const passwordPolicies = await getAllPasswordPoliciesRaw(domain.id, accessToken);
+    expect(passwordPolicies.raw.status).toEqual(204);
+  });
+
+  it('must create password policy', async () => {
+    const passwordPolicy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'default',
+    });
+    expect(passwordPolicy).toBeDefined();
+    expect(passwordPolicy.id).toEqual(passwordPolicy.id);
+    expect(passwordPolicy.defaultPolicy).toBeTruthy();
+  });
+
+  it('must update password policy', async () => {
+    const oldPasswordPolicy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'new',
+    });
+    const updatedPasswordPolicy = await updatePasswordPolicy(domain.id, accessToken, oldPasswordPolicy.id, {
+      name: 'updated-new',
+      defaultPolicy: true,
+    });
+    expect(updatedPasswordPolicy).toBeDefined();
+    expect(updatedPasswordPolicy.id).toEqual(updatedPasswordPolicy.id);
+    expect(updatedPasswordPolicy.defaultPolicy).toBeTruthy();
```

Comment:
**suggestion**: assertion on the name attribute could be nice and maybe doing a GET using the ID can be relevant to be sure the update really succeed

Reply:
fixed